Flaky test TestKuduIntegrationDecimalColumns. testDeleteByPrimaryKeyDecimalColumn

https://github.com/trinodb/trino/runs/2129444090
```
Error:  Tests run: 227, Failures: 1, Errors: 0, Skipped: 32, Time elapsed: 679.51 s <<< FAILURE! - in TestSuite
Error:  testDeleteByPrimaryKeyDecimalColumn(io.trino.plugin.kudu.TestKuduIntegrationDecimalColumns)  Time elapsed: 0.507 s  <<< FAILURE!
java.lang.RuntimeException: line 1:32: Column 'decimal_id' cannot be resolved
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:120)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:491)
	at io.trino.testing.QueryAssertions.assertUpdate(QueryAssertions.java:71)
	at io.trino.testing.AbstractTestQueryFramework.assertUpdate(AbstractTestQueryFramework.java:235)
	at io.trino.testing.AbstractTestQueryFramework.assertUpdate(AbstractTestQueryFramework.java:230)
	at io.trino.plugin.kudu.TestKuduIntegrationDecimalColumns.testDeleteByPrimaryKeyDecimalColumn(TestKuduIntegrationDecimalColumns.java:95)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.trino.spi.TrinoException: line 1:32: Column 'decimal_id' cannot be resolved
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48)
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43)
	at io.trino.sql.analyzer.SemanticExceptions.missingAttributeException(SemanticExceptions.java:33)
	at io.trino.sql.analyzer.Scope.lambda$resolveField$7(Scope.java:228)
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at io.trino.sql.analyzer.Scope.resolveField(Scope.java:228)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.visitIdentifier(ExpressionAnalyzer.java:485)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.visitIdentifier(ExpressionAnalyzer.java:404)
	at io.trino.sql.tree.Identifier.accept(Identifier.java:72)
	at io.trino.sql.tree.StackableAstVisitor.process(StackableAstVisitor.java:27)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.process(ExpressionAnalyzer.java:427)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.getOperator(ExpressionAnalyzer.java:1750)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.visitComparisonExpression(ExpressionAnalyzer.java:600)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.visitComparisonExpression(ExpressionAnalyzer.java:404)
	at io.trino.sql.tree.ComparisonExpression.accept(ComparisonExpression.java:71)
	at io.trino.sql.tree.StackableAstVisitor.process(StackableAstVisitor.java:27)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.process(ExpressionAnalyzer.java:427)
	at io.trino.sql.analyzer.ExpressionAnalyzer.analyze(ExpressionAnalyzer.java:359)
	at io.trino.sql.analyzer.ExpressionAnalyzer.analyzeExpression(ExpressionAnalyzer.java:1973)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.analyzeExpression(StatementAnalyzer.java:2925)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.analyzeWhere(StatementAnalyzer.java:2758)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.lambda$visitDelete$7(StatementAnalyzer.java:640)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitDelete(StatementAnalyzer.java:640)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitDelete(StatementAnalyzer.java:343)
	at io.trino.sql.tree.Delete.accept(Delete.java:61)
	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:360)
	at io.trino.sql.analyzer.StatementAnalyzer.analyze(StatementAnalyzer.java:323)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:91)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:83)
	at io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:263)
	at io.trino.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:186)
	at io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:768)
	at io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:129)
	at io.trino.$gen.Trino_testversion____20210317_095608_2849.call(Unknown Source)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

--- a/plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java
+++ b/plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java
@@ -101,15 +101,18 @@ public class TestKuduIntegrationDecimalColumns
     private void doTestCreateTableWithDecimalColumn(TestDecimal decimal)
     {
         String tableName = decimal.getTableName();
-        String dropTable = "DROP TABLE IF EXISTS " + tableName;
-        String createTable = "" +
-                "CREATE TABLE " + tableName + " (\n" +
-                "  id INT WITH (primary_key=true),\n" +
-                "  dec DECIMAL(" + decimal.precision + "," + decimal.scale + ")\n" +
+        String dropTable = format("DROP TABLE IF EXISTS %s", tableName);
+        String createTable = format(
+                "CREATE TABLE %s (\n" +
+                "   id INT WITH (primary_key=true),\n" +
+                "   dec DECIMAL(%s, %s)\n" +
                 ") WITH (\n" +
-                " partition_by_hash_columns = ARRAY['id'],\n" +
-                " partition_by_hash_buckets = 2\n" +
-                ")";
+                "   partition_by_hash_columns = ARRAY['id'],\n" +
+                "   partition_by_hash_buckets = 2\n" +
+                ")",
+                tableName,
+                decimal.precision,
+                decimal.scale);
 
         assertUpdate(dropTable);
         assertUpdate(createTable);
@@ -118,9 +121,9 @@ public class TestKuduIntegrationDecimalColumns
         int maxScale = decimal.precision - 10;
         int valuePrecision = decimal.precision - maxScale + Math.min(maxScale, decimal.scale);
         String insertValue = fullPrecisionValue.substring(0, valuePrecision + 1);
-        assertUpdate("INSERT INTO " + tableName + " VALUES(1, DECIMAL '" + insertValue + "')", 1);
+        assertUpdate(format("INSERT INTO %s VALUES(1, DECIMAL '%s')", tableName, insertValue), 1);
 
-        MaterializedResult result = computeActual("SELECT id, CAST((dec - (DECIMAL '" + insertValue + "')) as DOUBLE) FROM " + tableName);
+        MaterializedResult result = computeActual(format("SELECT id, CAST((dec - (DECIMAL '%s')) as DOUBLE) FROM %s", insertValue, tableName));
         assertEquals(result.getRowCount(), 1);
         Object obj = result.getMaterializedRows().get(0).getField(1);
         assertTrue(obj instanceof Double);

--- a/plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java
+++ b/plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java
@@ -16,6 +16,8 @@ package io.trino.plugin.kudu;
 import io.trino.testing.AbstractTestQueryFramework;
 import io.trino.testing.MaterializedResult;
 import io.trino.testing.QueryRunner;
+import io.trino.testing.sql.TestTable;
+import io.trino.testing.sql.TrinoSqlExecutor;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.Test;
 
@@ -65,70 +67,54 @@ public class TestKuduIntegrationDecimalColumns
     @Test
     public void testDecimalColumn()
     {
-        assertUpdate("CREATE TABLE IF NOT EXISTS test_decimal (" +
-                "id INT WITH (primary_key=true), " +
-                "col_decimal decimal(10, 6)" +
-                ") WITH (" +
-                " partition_by_hash_columns = ARRAY['id'], " +
-                " partition_by_hash_buckets = 2" +
-                ")");
-
-        assertUpdate("INSERT INTO test_decimal VALUES (0, 0.0), (2, 2.2), (1, 1.1)", 3);
-        assertQuery("SELECT * FROM test_decimal WHERE col_decimal = 1.1", "VALUES (1, 1.1)");
-        assertUpdate("DELETE FROM test_decimal WHERE col_decimal = 1.1", 1);
-
-        assertUpdate("DROP TABLE test_decimal");
+        try (TestTable testTable = new TestTable(
+                new TrinoSqlExecutor(getQueryRunner()),
+                "test_decimal",
+                "(id INT WITH (primary_key=true), col_decimal decimal(10, 6)) " +
+                        "WITH (partition_by_hash_columns = ARRAY['id'], partition_by_hash_buckets = 2)")) {
+            assertUpdate(format("INSERT INTO %s VALUES (0, 0.0), (2, 2.2), (1, 1.1)", testTable.getName()), 3);
+            assertQuery(format("SELECT * FROM %s WHERE col_decimal = 1.1", testTable.getName()), "VALUES (1, 1.1)");
+            assertUpdate(format("DELETE FROM %s WHERE col_decimal = 1.1", testTable.getName()), 1);
+            assertQueryReturnsEmptyResult(format("SELECT * FROM %s WHERE col_decimal = 1.1", testTable.getName()));
+        }
     }
 
     @Test
     public void testDeleteByPrimaryKeyDecimalColumn()
     {
-        assertUpdate("CREATE TABLE IF NOT EXISTS test_decimal (" +
-                "decimal_id decimal(18, 3) WITH (primary_key=true), " +
-                "col_decimal decimal(18, 3)" +
-                ") WITH (" +
-                " partition_by_hash_columns = ARRAY['decimal_id'], " +
-                " partition_by_hash_buckets = 2" +
-                ")");
-
-        assertUpdate("INSERT INTO test_decimal VALUES (1.1, 1.1), (2.2, 2.2)", 2);
-        assertUpdate("DELETE FROM test_decimal WHERE decimal_id = 2.2", 1);
-        assertQuery("SELECT * FROM test_decimal", "VALUES (1.1, 1.1)");
-
-        assertUpdate("DROP TABLE test_decimal");
+        try (TestTable testTable = new TestTable(
+                new TrinoSqlExecutor(getQueryRunner()),
+                "test_decimal",
+                "(decimal_id decimal(18, 3) WITH (primary_key=true), col_decimal decimal(18, 3)) " +
+                        "WITH (partition_by_hash_columns = ARRAY['decimal_id'], partition_by_hash_buckets = 2)")) {
+            assertUpdate(format("INSERT INTO %s VALUES (1.1, 1.1), (2.2, 2.2)", testTable.getName()), 2);
+            assertUpdate(format("DELETE FROM %s WHERE decimal_id = 2.2", testTable.getName()), 1);
+            assertQuery(format("SELECT * FROM %s", testTable.getName()), "VALUES (1.1, 1.1)");
+        }
     }
 
     private void doTestCreateTableWithDecimalColumn(TestDecimal decimal)
     {
-        String tableName = decimal.getTableName();
-        String dropTable = format("DROP TABLE IF EXISTS %s", tableName);
-        String createTable = format(
-                "CREATE TABLE %s (\n" +
-                "   id INT WITH (primary_key=true),\n" +
-                "   dec DECIMAL(%s, %s)\n" +
-                ") WITH (\n" +
-                "   partition_by_hash_columns = ARRAY['id'],\n" +
-                "   partition_by_hash_buckets = 2\n" +
-                ")",
-                tableName,
+        String tableDefinition = format(
+                "(id INT WITH (primary_key=true), dec DECIMAL(%s, %s)) " +
+                "WITH (partition_by_hash_columns = ARRAY['id'], partition_by_hash_buckets = 2)",
                 decimal.precision,
                 decimal.scale);
 
-        assertUpdate(dropTable);
-        assertUpdate(createTable);
+        try (TestTable testTable = new TestTable(new TrinoSqlExecutor(getQueryRunner()), decimal.getTableName(), tableDefinition)) {
+            String fullPrecisionValue = "1234567890.1234567890123456789012345678";
+            int maxScale = decimal.precision - 10;
+            int valuePrecision = decimal.precision - maxScale + Math.min(maxScale, decimal.scale);
+            String insertValue = fullPrecisionValue.substring(0, valuePrecision + 1);
+            assertUpdate(format("INSERT INTO %s VALUES(1, DECIMAL '%s')", testTable.getName(), insertValue), 1);
 
-        String fullPrecisionValue = "1234567890.1234567890123456789012345678";
-        int maxScale = decimal.precision - 10;
-        int valuePrecision = decimal.precision - maxScale + Math.min(maxScale, decimal.scale);
-        String insertValue = fullPrecisionValue.substring(0, valuePrecision + 1);
-        assertUpdate(format("INSERT INTO %s VALUES(1, DECIMAL '%s')", tableName, insertValue), 1);
-
-        MaterializedResult result = computeActual(format("SELECT id, CAST((dec - (DECIMAL '%s')) as DOUBLE) FROM %s", insertValue, tableName));
-        assertEquals(result.getRowCount(), 1);
-        Object obj = result.getMaterializedRows().get(0).getField(1);
-        assertTrue(obj instanceof Double);
-        Double actual = (Double) obj;
-        assertEquals(0, actual, 0.3 * Math.pow(0.1, decimal.scale), "p=" + decimal.precision + ",s=" + decimal.scale + " => " + actual + ",insert = " + insertValue);
+            MaterializedResult result = computeActual(format("SELECT id, CAST((dec - (DECIMAL '%s')) as DOUBLE) FROM %s", insertValue, testTable.getName()));
+            assertEquals(result.getRowCount(), 1);
+            Object obj = result.getMaterializedRows().get(0).getField(1);
+            assertTrue(obj instanceof Double);
+            Double actual = (Double) obj;
+            assertEquals(0, actual, 0.3 * Math.pow(0.1, decimal.scale), "p=" + decimal.precision + ",s=" + decimal.scale + " => " + actual + ",insert = " + insertValue);
+        }
     }
 
     static class TestDecimal

